test/SourceKit/Indexing/index_effective_access_level.swift

```diff
@@ -0,0 +1,71 @@
+// RUN: %sourcekitd-test -req=index %s -- -Xfrontend -serialize-diagnostics-path -Xfrontend %t.dia %s | %sed_clean > %t.response
+// RUN: %diff -u %s.response %t.response
+
+public enum PublicEn {
+    case a
+    case b
+}
+
+enum InternalEn {
+    case a
+    case b
+}
+
+fileprivate enum FilePrivateEn {
+    case a
+    case b
+}
+
+private enum PrivateEn {
+    case a
+    case b
+}
+
+extension PublicEn {
+    public func puFoo() {}
+}
+
+public extension PublicEn {
+    func puFooFromPublicExtension() {}
+}
+
+extension InternalEn {
+    func foo() {}
+}
+
+extension FilePrivateEn {
+    fileprivate func flPrFoo() {}
+}
+
+fileprivate extension FilePrivateEn {
+    func flPrFooFromFilePrivateExtension() {}
+}
+
+extension PrivateEn {
+    private func prFoo() {}
+}
+
+private extension PrivateEn {
+    func prFooFromPrivateExtension() {}
+}
+
+private struct ScopeReducerStruct {
+    public func a() {}
+    func b() {}
+    fileprivate func c() {}
+    private func d() {}
+}
+
+public struct ScopeKeeperStruct {
+    public func a() {}
+    func b() {}
+    fileprivate func c() {}
+    private func d() {}
+}
+
+struct PartialScopeReducerStruct {
+    public func a() {}
+    func b() {}
+    fileprivate func c() {}
+    private func d() {}
+}
```

Comment:
These test cases are great. One nit and one suggestion:

* Nit: Would you mind giving these a consistent naming scheme? It would just make it easier to review the result. Perhaps something like `PrivateEnum`, `PrivateStruct`, `InternalEnum`, `InternalStruct`, `filePrivateFunc`, `publicFunc`, etc. (The cases are fine as-is, since they don't have any access modifiers, but perhaps even they can be given unique letters.) That way, it'll be easier for a reader (i.e., me) to pick out when the results are unexpected. Also, it can be helpful to readers like me to avoid ad-hoc abbreviations like `flPr`.

* Suggestion: Could you please add (a) some `private` extensions with `private` members and `public` extensions with members that don't have any access modifier; (b) some declarations other than cases and functions (initializers, subscripts, variables with different getters and setters); (c) protocol declarations, since the members can't have their own access modifiers? Should all work but just in case...